pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -316,9 +316,9 @@ func (k *KSMCheck) processMetrics(sender aggregator.Sender, metrics map[string][
 				// Some metrics can be aggregated and consumed as-is or by a transformer.
 				// So, let’s continue the processing.
 			}
-			if metadataMetricsRegex.MatchString(metricFamily.Name) {
+			if _, found := metricNamesMapper[metricFamily.Name]; !found && metadataMetricsRegex.MatchString(metricFamily.Name) {
```

Comment:
I don't think we need this anymore

Reply:
Technically we don't, but I believe that's the way it should work, may prevent a lengthy investigation in the future.

Reply:
no strong opinion but we have Line 343 for a reason `log.Tracef("KSM metric '%s' is unknown for the check, ignoring it", metricFamily.Name)`

Reply:
I think it's not the same. If I remove this code and someone wants to add a metric matching the metadata pattern `".*_(info|labels|status_reason)"`, line 343 would not log and the added metric won't show-up without any indication about why.

Reply:
ok, in this case we need to check whether `metricTransformers` contains the metric as well

Reply:
for completeness

Reply:
Agreed, in fact I just need to move the check at the bottom to make it work as expected.